### PR TITLE
Docs: Add Vue syntax highlighting

### PR DESCRIPTION
**Pull request by hinogi**
Added vue syntax highligh support

Closes #

#### The changes in this PR are covered in the following automated tests:

- [ ] stories
- [ ] unit tests
- [ ] integration tests
- [ ] end-to-end tests

#### Manual testing

_This section is mandatory for all contributions. If you believe no manual test is necessary, please state so explicitly. Thanks!_

### Documentation

- [ ] Add or update documentation reflecting your changes
- [ ] If you are deprecating/removing a feature, make sure to update
      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

## Checklist for Maintainers

- [ ] When this PR is ready for testing, make sure to add `ci:normal`, `ci:merged` or `ci:daily` GH label to it to run a specific set of sandboxes. The particular set of sandboxes can be found in `code/lib/cli-storybook/src/sandbox-templates.ts`
- [ ] Make sure this PR contains **one** of the labels below:
   <details>
     <summary>Available labels</summary>

  - `bug`: Internal changes that fixes incorrect behavior.
  - `maintenance`: User-facing maintenance tasks.
  - `dependencies`: Upgrading (sometimes downgrading) dependencies.
  - `build`: Internal-facing build tooling & test updates. Will not show up in release changelog.
  - `cleanup`: Minor cleanup style change. Will not show up in release changelog.
  - `documentation`: Documentation **only** changes. Will not show up in release changelog.
  - `feature request`: Introducing a new feature.
  - `BREAKING CHANGE`: Changes that break compatibility in some way with current major version.
  - `other`: Changes that don't fit in the above categories.

   </details>

### 🦋 Canary release

This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_

## Greptile Summary

Added Vue syntax highlighting support to the SyntaxHighlighter component by importing and registering Vue language highlighting from react-syntax-highlighter's Highlight.js package.

- Added Vue language import from `react-syntax-highlighter/dist/esm/languages/hljs/vue` in `code/core/src/components/components/syntaxhighlighter/syntaxhighlighter.tsx`
- Potential compatibility issue: Vue uses Highlight.js while other languages use Prism, which may cause inconsistent highlighting
- Added Vue to supported languages object in SyntaxHighlighter component

### CODE REVIEW

File: code/core/src/components/components/syntaxhighlighter/syntaxhighlighter.tsx [hunk 1]
```diff
@@ -46,6 +47,7 @@ export const supportedLanguages = {
   tsx,
   typescript,
   graphql,
+  vue
```
[the remark this answers is not in the document]
Comment (hinogi) [reply code/core/src/components/components/syntaxhighlighter/syntaxhighlighter.tsx:50]:
```suggestion
  vue,
```

File: code/core/src/components/components/syntaxhighlighter/syntaxhighlighter.tsx [hunk 2]
```diff
@@ -22,6 +22,7 @@ import tsx from 'react-syntax-highlighter/dist/esm/languages/prism/tsx';
 import typescript from 'react-syntax-highlighter/dist/esm/languages/prism/typescript';
 import yml from 'react-syntax-highlighter/dist/esm/languages/prism/yaml';
 import ReactSyntaxHighlighter from 'react-syntax-highlighter/dist/esm/prism-light';
+import vue from 'react-syntax-highlighter/dist/esm/languages/hljs/vue';
```
[the remark this answers is not in the document]
Comment (hinogi) [reply code/core/src/components/components/syntaxhighlighter/syntaxhighlighter.tsx:25]: image
no prism vue available

[the remark this answers is not in the document]
Comment (hinogi) [reply code/core/src/components/components/syntaxhighlighter/syntaxhighlighter.tsx:25]: image

**Review by kasperpeulen — APPROVED** [review]
@nde

File: code/core/src/components/components/syntaxhighlighter/syntaxhighlighter.tsx (hunk 2 above)
Comment (kasperpeulen) [inline code/core/src/components/components/syntaxhighlighter/syntaxhighlighter.tsx:25]: This import doesn't seem to exist (see the build failure.

**Review by kasperpeulen — CHANGES_REQUESTED** [review]
The export doesn't seem to exist

File: code/core/src/components/components/syntaxhighlighter/syntaxhighlighter.tsx (hunk 2 above)
Comment (hinogi) [reply code/core/src/components/components/syntaxhighlighter/syntaxhighlighter.tsx:25]: Oh that is my fault, I assumed it uses highlightjs

**Pull request closed without merging**